internal/wclayer/cim/process.go

```diff
@@ -22,77 +22,14 @@ import (
-		if err := os.RemoveAll(baseVhdPath); err != nil {
-			return fmt.Errorf("failed to remove base vhdx path:  %w", err)
-		}
-	}
-	if _, err := os.Stat(diffVhdPath); err == nil {
-		if err := os.RemoveAll(diffVhdPath); err != nil {
-			return fmt.Errorf("failed to remove differencing vhdx: %w", err)
-		}
-	}
-
-	createParams := &vhd.CreateVirtualDiskParameters{
-		Version: 2,
-		Version2: vhd.CreateVersion2{
-			MaximumSize:      defaultVhdSize * memory.GiB,
-			BlockSizeInBytes: defaultVHDXBlockSizeInMB * memory.MiB,
-		},
-	}
-	handle, err := vhd.CreateVirtualDisk(baseVhdPath, vhd.VirtualDiskAccessNone, vhd.CreateVirtualDiskFlagNone, createParams)
-	if err != nil {
-		return fmt.Errorf("failed to create vhdx: %w", err)
-	}
-
-	defer func() {
-		if err != nil {
-			os.RemoveAll(baseVhdPath)
-			os.RemoveAll(diffVhdPath)
-		}
-	}()
-
-	err = computestorage.FormatWritableLayerVhd(ctx, windows.Handle(handle))
-	// we always wanna close the handle whether format succeeds for not.
-	closeErr := syscall.CloseHandle(handle)
-	if err != nil {
-		return err
-	} else if closeErr != nil {
-		return fmt.Errorf("failed to close vhdx handle: %w", closeErr)
-	}
-
-	// Create the differencing disk that will be what's copied for the final rw layer
-	// for a container.
-	if err = vhd.CreateDiffVhd(diffVhdPath, baseVhdPath, defaultVHDXBlockSizeInMB); err != nil {
-		return fmt.Errorf("failed to create differencing disk: %w", err)
-	}
-
-	if err = security.GrantVmGroupAccess(baseVhdPath); err != nil {
-		return fmt.Errorf("failed to grant vm group access to %s: %w", baseVhdPath, err)
-	}
-	if err = security.GrantVmGroupAccess(diffVhdPath); err != nil {
-		return fmt.Errorf("failed to grant vm group access to %s: %w", diffVhdPath, err)
-	}
-	return nil
-}
-
-// processUtilityVMLayer is similar to createContainerBaseLayerVHDs but along with the scratch creation it
-// also does some BCD modifications to allow the UVM to boot from the CIM. It expects that the UVM BCD file is
-// present at layerPath/`wclayer.BcdFilePath` and a UVM SYSTEM hive is present at
-// layerPath/UtilityVM/`wclayer.RegFilesPath`/SYSTEM. The scratch VHDs are created under the `layerPath`
-// directory.
+// processUtilityVMLayer creates a base VHD for the UtilityVM's scratch. Configures the BCD file at path
+// "layerPath/`wclayer.BcdFilePath`" to make the UVM boot from this base VHD.  Also, configures the UVM's
+// SYSTEM hive at path "layerPath/UtilityVM/`wclayer.RegFilesPath`/SYSTEM" to specify that the UVM is booting
+// from a CIM.
 func processUtilityVMLayer(ctx context.Context, layerPath string) error {
-	// func createUtilityVMLayerVHDs(ctx context.Context, layerPath string) error {
-	baseVhdPath := filepath.Join(layerPath, wclayer.UtilityVMPath, wclayer.UtilityVMBaseVhd)
-	diffVhdPath := filepath.Join(layerPath, wclayer.UtilityVMPath, wclayer.UtilityVMScratchVhd)
+	baseVhdPath := filepath.Join(layerPath, wclayer.UtilityVMPath, wclayer.UtilityVMScratchVhd)
```

Comment:
Based on previous discussion I thought you were going to change this back to creating a diff VHD?

Reply:
Yes, should be fixed now. 
Diff VHD creation for container is done in the snapshotter here: https://github.com/containerd/containerd/pull/9659/files#diff-082dce0a0ee622ef2cc238f2e296ddc3a7a9c6c4af0be76295cc571352cd27a6